Tests/NIOTests/SocketChannelTest.swift

```diff
@@ -322,16 +322,15 @@ public class SocketChannelTest : XCTestCase {
         let clientSock = try Socket(protocolFamily: AF_INET, type: Posix.SOCK_STREAM)
         let connected = try clientSock.connect(to: serverChannel.localAddress!)
         XCTAssertEqual(connected, true)
-        let clientChannelFuture = try clientSock.withUnsafeFileDescriptor {
+        let clientChannel = try clientSock.withUnsafeFileDescriptor {
             ClientBootstrap(group: group).withConnectedSocket(descriptor: dup($0))
-        }
+        }.wait()
         try clientSock.close()
-        let clientChannel = try clientChannelFuture.wait()
```

Comment:
This fairly substantially changes the semantic of this test, why are we doing it?

Reply:
@Lukasa I can revert it if you want but for the test it not matters really...

Reply:
I just thought it makes sense to remove local vars that are not really needed.

Reply:
It changes the behaviour quite dramatically. I'm not sure the behavioural change is worse, it may in fact be better (as right now the `close` races with the `withConnectedSocket` call). But it does change the behaviour.

Reply:
yeah... maybe we should loop in @vlm ... WDYT @vlm  ?

Reply:
it does change behaviour so I'm not sure about this one but if @vlm is happy, I'm too